src/Stack/Application.php

```diff
@@ -97,8 +100,31 @@ protected function buildResponseFromResponseBag(ResponseBag $responseBag)
         $result = $responseBag->getResult();
         $resultCode = $responseBag->getResultCode();
         $headers = $responseBag->getHeaders();
+        $type = $responseBag->getType();
 
-        return new Response($result, $resultCode, $headers);
+        if ($type === ResponseBag::TYPE_DEFAULT) {
```

Comment:
should retrieve the response object through a factory, so that you don't hardcode available types in application